### PR TITLE
added SignUp page & react router

### PR DESCRIPTION
**Pull request by BesoBerdzenishvili**
SignUp page on /signup path is added.

This page allows registration of new members. It asks for username, full name, email and password. 

userName checks if input is alphanumeric.

Axios is used to post data on server. I send post request on my server and it works.
Error message is also displayed from this function on the top of form field (in case there is any error).

React router is also added, so we wouldn't need to use server for routing (because it's slower).

SignUp component has styled 'form' and 'i' elements. 'Form' is positioned at the center of screen. 'I' elements is for error message and text is displayed in red;

### CODE REVIEW

**Comment by ggogobera** [comment]
> React router is also added, so we wouldn't need to use server for routing (because it's slower).

This isn't always the case. Check how fast is RemixJS w/ form actions and w/o JS.

> Axios is used to post data on server

There was no such requirement in the issue. Why not `fetch`?

**Review by ggogobera — CHANGES_REQUESTED** [review]
To be honest, overall it's good, but you make the same mistakes and this isn't productive.

Either put stuff onto notes or we ain't going nowhere like this.

File: src/components/pages/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const [fullName, setFullName] = useState('')
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [confirmPassword, setconfirmPassword] = useState('')
+
+  const handleSubmit = useCallback(
+    async (e) => {
+      e.preventDefault()
+      try {
+        await fetch('http://localhost:5000/api/register', {
```
Comment (ggogobera) [inline src/components/pages/SignUp.js:35]: How do you validate if a server returned 4xx status code? 

For example, if `userName` is already taken, the server (that we will connect to) will return 400 status code indicating that `userName` is taken.

**Review by ggogobera — APPROVED** [review]
Looks almost perfect! 👍🏼

I've left one last nit comment.

File: src/components/pages/SignUp.js (hunk 1 above)
Comment (BesoBerdzenishvili) [reply src/components/pages/SignUp.js:35]: Okay, what do you think about this algorithm:
1) create a function "validate" that takes userName, fullName, email, password, confirmPassword and checks if:
confirmPassword and password match;
password is at least 6 characters long;
email is at least 5 characters long, has "@" symbol and a dot;
userName doesn't exist in a database;
and that all fields are filled; 
2) give each case it's error message e.g. "password must contain at leas 6 symbols"
3) create: let error = [];
4) if any of the above cases is true push error message in 'error'
5) if error.length >= 1 don't execute fetch request and display error

Comment (ggogobera) [reply src/components/pages/SignUp.js:35]: First of all, never say "algorithm" 😂. It's not an algorithm.

I'm not asking to comment on that here. I'm asking to add the logic in the PR.

But, you are explaining the server logic. I'm not interested in server logic; it's already working. I'm asking you to handle the response from the server here.

**Review by ggogobera — CHANGES_REQUESTED** [review]
This is not what I'm asking for.

I'm saying that server returns 400 status code with data.

Handle returned `data`.

This commit just sets error in the state if `try` block throws some error.

**Review by ggogobera — APPROVED** [review]

**Pull request merged**